pkg/api/query/v1.go

```diff
@@ -439,16 +440,39 @@ func (qapi *QueryAPI) labelValues(r *http.Request) (interface{}, []error, *api.A
 		return nil, nil, apiErr
 	}
 
-	q, err := qapi.queryableCreate(true, nil, storeDebugMatchers, 0, enablePartialResponse, false).
+	var matcherSets [][]*labels.Matcher
+	for _, s := range r.Form[MatcherParam] {
+		matchers, err := parser.ParseMetricSelector(s)
+		if err != nil {
+			return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: err}
+		}
+		matcherSets = append(matcherSets, matchers)
+	}
+
+	q, err := qapi.queryableCreate(true, nil, storeDebugMatchers, 0, enablePartialResponse, true).
```

Comment:
Not a nit, just wanted to ask a question. Why did you flip the last argument (`skipChunks`) to `true`, and what it does exactly?

Reply:
A query result contains two parts, the series metadata like label sets, and the sample values. In this case, since we are just querying labels, we can set `skipChunks` to not return samples, only labels are fine.

Reply:
The original issue about this option https://github.com/thanos-io/thanos/issues/1822. 
Also this Prometheus pr https://github.com/prometheus/prometheus/pull/8050 has the same idea to improve the performance.

Reply:
Cool, thanks for the answer. :)